Mathlib/Logic/Denumerable.lean

```diff
@@ -189,10 +189,9 @@ variable {s : Set ℕ} [Infinite s]
 
 section Classical
 
-open scoped Classical
-
-theorem exists_succ (x : s) : ∃ n, (x : ℕ) + n + 1 ∈ s :=
-  _root_.by_contradiction fun h =>
+theorem exists_succ (x : s) : ∃ n, (x : ℕ) + n + 1 ∈ s := by
+  classical
+  exact _root_.by_contradiction fun h =>
```

Comment:
Can this be optimized using `by_contra h` or something like that?

Reply:
Yes, it can - pushed.